src/main/java/org/betonquest/betonquest/commands/QuestCommand.java

```diff
@@ -1658,27 +1668,19 @@ private void handleDebug(final CommandSender sender, final String... args) {
                 LOG.debug("Cannot continue, sender must be player");
                 return;
             }
-            final PlayerFilter playerHandler = BetonQuest.getInstance().getLogWatcher().getPlayerFilter();
             final UUID uuid = ((Player) sender).getUniqueId();
             if (args.length < 3) {
-                sender.sendMessage("§2Active Filters: " + String.join(", ", playerHandler.getFilters(uuid)));
+                sender.sendMessage("§2Active Filters: " + String.join(", ", logWatcher.getActivePatterns(uuid)));
                 return;
             }
             final String filter = args[2];
-            final boolean exist = playerHandler.getFilters(uuid).contains(filter);
-            if (exist && args.length == 3) {
-                if (!playerHandler.removeFilter(uuid, filter)) {
-                    sender.sendMessage("§2Filter could not be removed!");
-                    return;
-                }
+            if (logWatcher.isActivePattern(uuid, filter) && args.length == 3) {
+                logWatcher.removeFilter(uuid, filter);
                 sender.sendMessage("§2Filter removed!");
                 return;
             }
             final Level level = getLogLevel(args.length > 3 ? args[3] : null);
-            if (!playerHandler.addFilter(uuid, filter, level)) {
-                sender.sendMessage("§2Filter could not be added!");
-                return;
-            }
```

Comment:
I am not sure if we may want to keep the information if the filter was changed/applied or not, because it already exists or because it is not an active filter. May improve the error message? But this is a general Question, not an issue itself